Kajabi/redis_dedupe_set
# [AOPS-66] Add a Helper method and remove member from Set on raised error

ℹ️  Most of the changes in here are due to slight refactoring. It might be easier to look commit-by-commit. The big refactor one is the very last.


### Jira Ticket
https://kajabi.atlassian.net/browse/AOPS-66


### Description
Nothing major is wrong here, but it would be nice to add a couple of conveniences...


### Solution
* add a new helper method `RedisDedupe::Set#max_member` to return the largest member existing in the set, so that in the case of unique integer identifiers, that can be used to limit an ActiveRecord query.
* remove the member from the set (the member just added to the set) if the block yielded to by `RedisDedupe::Set#check` raises an error
* updated the gem; added and fixed up for rubocop


### Customer Impact
n/a


### QA Testing Guidelines
Can fire up a console and run some code there... see the specs for examples.
This won't be used until the version is specified in the main kajabi app.


### Checklist
- [x] Did you add or update specs as appropriate?
  - yes
- [x] Are the changes behind a feature flag? (If applicable)
  - n/a
- [x] Is there any documentation needed? (Tooltips, Help Article links, etc.)
  - no
- [x] Did you add or edit a controller endpoint?
  - no
    - n/a
